doc/program_services.md

```diff
@@ -0,0 +1,1124 @@
+# Program Services
+
+## Save area and linkage conventions
+
+Standard save-area is defined as follows:
+
+``` hlasm
+         DS   0CL72
+         DS   F         +0 reserved
+         DS   F         +4 address of callers save-area
+         DS   F         +8 address of our save-area
+         DS   15F       +12 callers GR14 through GR12
+```
+
+A program is invoked with entry point in GR15 and return address in GR14.
+
+GR15 is expected to contain a return code upon exit by convention.
+
+GR13 conventionally points to our save area.
+
+## Passing parameters to the initial program
+
+There are two methods of passing parameters to a program
+
+* SYSPARM on mz390
+* PARM on ez390
+
+In either case enclose the entire parm to be passed in double quotes.
+The double quotes are required to handle commas and spaces that otherwise cause command processors to split the parm. 
+The double quotes are not required if there are no commas or spaces in the text. 
+
+### SYSPARM
+
+If single quotes are included in text, they are passed on to &SYSPARM.
+
+There is a limit of 32767 bytes for the text. 
+
+SYSPARM will be transferred to the macro variable &SYSPARM.
+
+``` dos
+mz390 ... "SYSPARM(HELLO WORLD)"
+```
+Access the text by coding:
+
+``` hlasm
+label    DC    C'&SYSPARM'
+...... will translate to 
+label    DC    C'HELLO WORLD'
+```
+
+### PARM
+
+PARM can be accessed via GR1 at program entry and consists of a halfword length followed by the text.
+
+``` dos
+ez390 ... "PARM(HELLO WORLD)"
+```
+
+GR1 points to:
+
+``` hlasm
+         DC   H'11',C'HELLO WORLD'`
+```
+
+If single quotes are included around text in PARM they are removed.
+
+``` dos
+ez390 .... "PARM('HELLO WORLD')" 
+```
+
+Will also result in GR1 pointing to:
+
+``` hlasm
+         DC   H'11',C'HELLO WORLD'`
```

Comment:
Line 53. Requires change. GR1 points to a fullword which points to a halfword followed by the parm value:

R1 -> word -> halfword -length-of-parm-value || parm-value

Sample code snippet that accesses parm value:
`        ...`
         L     R2,0(,R1)           R2 --> PARM=value area
         LH    R3,0(,R2)           R3  =  length of value
         LTR   R3,R3               Any value?
         BZ    TADflt              No; use default
         CHI   R3,3                More than 3 digits?
         BH    TADflt              Yes; use default
         ...
TAPACK   PACK  DW,2(*-*,R2)        Convert unpacked to packed`

Reply:
Have made some edits in b0a360f. Let me know if you think this addresses the issue.